src/main/java/com/amazon/awslabs/cloudfoody/OpenTable.java

```diff
@@ -0,0 +1,145 @@
+package com.amazon.awslabs.cloudfoody;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class OpenTable {
+
+    private static final String BASE_URL = "http://opentable.herokuapp.com/api/";
+
+    /**
+     *  The method establishes a connection and returns a url object
+     */
+    public static URLConnection createConnection(String urlString) throws IOException {
+        URL url = new URL(urlString);
+        return url.openConnection();
+    }
+
+    /**
+     *  The method takes a restaurant name as an arguement and returns back an instance of the Restaurant class
+     *  and sets all the fields as well
+     */
+    public static Restaurant getRestaurantByName(String name) throws IOException {
+        String fullURL = BASE_URL + "restaurants?name=" + name;
+        URLConnection connection = createConnection(fullURL);
```

Comment:
Please try this:

        final ObjectMapper mapper = new ObjectMapper();
        final InputStreamReader streamReader = new InputStreamReader(connection.getInputStream());
        
        return mapper.readValue(streamReader, Restaurant.class);